source/Ship.cpp

```diff
@@ -2280,6 +2280,32 @@ bool Ship::IsOverheated() const
 
 
 
+bool Ship::IsIonized() const
+{
+	if(!ionization)
+		return false;
+
+	// A ship can only be fully ionized if its engines or weapons require energy.
+	bool movementEnergy = attributes.Get("thrusting energy") > 0
+		|| attributes.Get("reverse thrusting energy") > 0
+		|| attributes.Get("turning energy") > 0;
+
+	bool firingEnergy = false;
+	for(const auto &it : outfits)
+	{
+		const Weapon *weapon = it.first->GetWeapon().get();
+		if(weapon && weapon->FiringEnergy() > 0)
+		{
+			firingEnergy = true;
+			break;
+		}
+	}
+
+	return movementEnergy || firingEnergy ? ionization > energy : false;
```

Comment:
We can actually simplify this further; we don't need to check the weapons if we already know that energy is needed for movement.
```suggestion
	bool usesEnergy = attributes.Get("thrusting energy") > 0
		|| attributes.Get("reverse thrusting energy") > 0
		|| attributes.Get("turning energy") > 0
		|| any_of(outfits.begin(), outfits.end(), [](const auto &it) -> bool {
			const Weapon *weapon = it.first->GetWeapon().get();
			return weapon && weapon->FiringEnergy() > 0;
		});

	return usesEnergy ? ionization > energy : false;
```